src/common/sensor.rs

```diff
@@ -0,0 +1,75 @@
+#![allow(dead_code)]
+
+use crate::common::status::Status;
+use std::collections::BTreeMap;
+use std::collections::HashMap;
+
+pub type GenericReadingsResult =
+    ::std::collections::HashMap<::prost::alloc::string::String, ::prost_types::Value>;
+
+pub type TypedReadingsResult<T> = ::std::collections::HashMap<String, T>;
```

Comment:
I don't particularly love this, but my attempts to make either a generic`Sensor` type or to give `Sensor` a generic `get_readings` did not succeed, primarily because it meant that the type parameter became required everywhere.